internal/guest/runtime/hcsv2/container.go

```diff
@@ -191,7 +197,18 @@ func (c *Container) Delete(ctx context.Context) error {
 		}
 	}
 
-	return c.container.Delete()
+	var retErr error
+	if err := c.container.Delete(); err != nil {
+		retErr = err
+	}
+
+	if err := os.RemoveAll(c.scratchDirPath); err != nil {
+		if retErr == nil {
+			retErr = err
+		}
+	}
```

Comment:
Could we concatenate the error msgs if both of these failed?

Reply:
Done.